### PR TITLE
[DOCS] Add top-level EQL docs page. Adds EQL requirements page.

### PR DESCRIPTION
**Pull request by jrodewig**
### Changes
* **Creates a top-level page for EQL in the ES reference.** 
   This page contains a high-level introduction and will include a nav for other EQL docs pages as they're built.

* **Creates a requirements page.**
  This page outlines the fields needed to use EQL in ES. 

### Note to reviewers
This PR isn't intended to encompass all needed EQL docs. I'll be working incrementally to add new pages. However, the overview page is a required first step for adding those new pages.

### CODE REVIEW

**Review by aleksmaus — APPROVED** [review]
LGTM, Thank you!

File: docs/reference/eql/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+[float]
+[[when-to-use-eql]]
+=== When to use EQL
+
+Consider using EQL if you:
+
+* Use {es} for threat hunting or other security use cases
+* Search time-series data or logs, such as network or system logs
+* Want an easy way to explore relationships between events
```
Comment (jrodewig) [inline docs/reference/eql/index.asciidoc]: @sethpayne Do you think this covers most of our major use cases for EQL within ES?

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
Thanks @aleksmaus!

**Comment by jrodewig** [comment]
### Backport commits
master  0a31b67
7.x 65f49d0bbae6403130dd69443995af0cc8b4b8df

**Pull request merged**